composer.json

```diff
@@ -18,9 +18,9 @@
         "ext-json": "*"
     },
     "require-dev": {
-        "phpunit/phpunit": "^7.5.15|^8.5",
+        "phpunit/phpunit": "^7.5.15 || ^8.5 || ^9.0",
         "php-coveralls/php-coveralls": "^2.2",
-        "vimeo/psalm": "^3.5"
+        "vimeo/psalm": "3.14.*"
```

Comment:
I fixed Psalm version here, to have more consistent builds over time. It should be upgraded sometimes, of course, but having stable builds is more important (as you correctly pointed).

Reply:
I think it might be worth going further and constraining to a specific point release of Psalm (e.g. current release ` 3.14.2`. I don't think it's very unusual for psalm patch level releases to detect errors that the previous release wouldn't have detected.

If we don't constrain it to the patch then someone making a PR may have the build fail because of an issue in code they haven't touched and shouldn't be expected to fix.